File: libs/document-processor/src/repo/core/schema/TriggerEvent.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId, blueNodeField } from '@blue-labs/language';

export const TriggerEventSchema = withTypeBlueId(blueIds['Trigger Event'])(
  z.object({
    event: blueNodeField(),
  })
);

export type TriggerEvent = z.infer<typeof TriggerEventSchema>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/TriggerEvent.java
package blue.language.types.conversation;

import blue.language.model.Node;
import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

@TypeAlias("Conversation/Trigger Event")
@TypeBlueId("GxUtWr3eH9a6YQeioQkujEnsPjD5s9RU8ZhEfmsV1vuU")
public class TriggerEvent {
    public Node event;

    public TriggerEvent event(Node event) {
        this.event = event;
        return this;
    }
}
